app/controllers/course/material/controller.rb

```diff
@@ -2,8 +2,8 @@ class Course::Material::Controller < Course::ComponentController
   load_and_authorize_resource :folder, through: :course, through_association: :material_folders,
                                        class: Course::Material::Folder.name
 
-  before_action :check_component
-  before_action :add_folder_breadcrumb
+  before_action :check_component, unless: :material_show
```

Comment:
I think should be  yes :) Because you know assessments depends on materials, and when materials is disabled, user should still be able to download assessment files( in assessment show page )